sourcecode/chesssimul/board.py

```diff
@@ -1,169 +1,182 @@
+        start_square: Square = self.board[move.start[0]][move.start[1]]
+        #Y a t'il une pièce à déplacer
+        if not (start_square.isEmpty()):
+            #type de la pièce que nous voulons déplacer
+            type_piece_start: PieceType = start_square.piece.kind
+            print(type_piece_start)
+            # Si c'est le cavalier
+            if (type_piece_start == PieceType.KNIGHT):
+                print("vérification du coup du cavalier")
+                self.move_knight(move)
+            if (type_piece_start == PieceType.BISHOP):
+                print("vérification du coup du fou")
+                self.move_bishop(move)
+
+
+
+    def move_knight(self, move: Move):
+        #postion de la pièce avant son déplacement dans l'échiquier
+        start_square: Square = self.board[move.start[0]][move.start[1]]
+        #postion de la pièce après son déplacement dans l'échiquier
+        end_square: Square = self.board[move.end[0]][move.end[1]]
+        if (move.end[0] == move.start[0] + 2 or move.end[0] == move.start[0] - 2):
+            if (move.end[1] == move.start[1] + 1 or move.end[1] == move.start[1] - 1):
+                if end_square.isEmpty:
+                    self.board[move.end[0]][move.end[1]].piece = start_square.piece
+                    self.board[move.start[0]][move.start[1]].piece = None
+                    print("cavalier déplacé")
+                elif not (end_square.isEmpty):
+                    if (end_square.piece.color != start_square.piece.color):
+                        self.board[move.end[0]][move.end[1]].piece = start_square.piece
+                        self.board[move.start[0]][move.start[1]].piece = None
+                        print("cavalier déplacé et prise pièce")
+                    else:
+                        print("impossible de prendre sa propre pièce")
+
+        elif (move.end[1] == move.start[1] + 2 or move.end[1] == move.start[1] - 2):
+            if (move.end[0] == move.start[0] + 1 or move.end[0] == move.start[0] - 1):
+                if not (end_square.isEmpty):
+                    if end_square.isEmpty:
+                        self.board[move.end[0]][move.end[1]].piece = start_square.piece
+                        self.board[move.start[0]][move.start[1]].piece = None
+                        print("cavalier déplacé")
+                elif not (end_square.isEmpty):
+                    if (end_square.piece.color != start_square.piece.color):
+                        self.board[move.end[0]][move.end[1]].piece = start_square.piece
+                        self.board[move.start[0]][move.start[1]].piece = None
+                        print("cavalier déplacé et prise pièce")
+                    else:
+                        print("impossible de prendre sa propre pièce")
+        else:
+            print("Impossible de déplacer le cavalier à cet endroit")
+
+
+    def move_bishop(self, move: Move):

```

Comment:
ditto check for board coordinates. Should probably be inside the make_move function in order for it to apply to all 6 pieces move functions.

Reply:
ditto return boolean